### ISSUE TITLE
SA1313 conflicts with IDE0060 recommendation

### ISSUE THREAD

**Issue by bbarry** [DotNetAnalyzers/StyleCopAnalyzers#3113]
Built in Roslyn IDE diagnostic "Unused Parameter" recommends using parameters `_` or `_#` to suppress the IDE diagnostic for public members where the parameter is unused but must exist for compatibility. 

I think that SA1313 should provide the same exception: if a parameter is unused and the parameter name starts with `"_"` and is optionally followed by a string of digits, SA1313 should not trigger.

for reference see Roslyn source:
https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

**Comment by sharwell** [comment]
This is related to (but not quite a duplicate of) #2599. I don't personally agree with the IDE0060 recommendation, and lean against making an exception to SA1313 for parameters.

**Comment by sharwell** [comment]
My personal recommendation is to add this on the first line of the method body:

```csharp
// Comment explaining why the parameter is included but not used
_ = parameterName;
```

**Comment by bbarry** [comment]
I think that recommendation could work. In my case where I am hitting it, I'm overriding comparison operators because another analyzer is making me do so (because I have to implement IComparable for the type to pass it to a 3rd party). The methods are:

        public static bool operator ==(Unit _1, Unit _2) => true;

        public static bool operator !=(Unit _1, Unit _2) => false;

        public static bool operator <(Unit _1, Unit _2) => false;

        public static bool operator <=(Unit _1, Unit _2) => true;

        public static bool operator >(Unit _1, Unit _2) => false;

        public static bool operator >=(Unit _1, Unit _2) => true;


An alternative solution to fix my specific problem would be to special case operators and maybe standardize their names?

**Comment by sharwell** [comment]
🔗 That's related to proposal #3046.
